sampling_kinematics_plugin/src/external_axis_sampling_kinematics_plugin.cpp

```diff
@@ -156,6 +158,32 @@ bool ExternalAxisSamplingKinematicsPlugin::initialize(const std::string& robot_d
     return false;
   }
 
+  // Get the cost weights
+  const std::string weight_param = "weights";
+  const std::size_t n = joint_model_group_->getActiveJointModelNames().size();
+  std::vector<double> tmp_weights;
+  if (!lookupParam(weight_param, tmp_weights, std::vector<double>(n, 1.0)))
+  {
+    ROS_WARN_STREAM_NAMED(LOG_NAMESPACE, "Using default weight vector (" << n << " x 1.0)");
+    cost_weights_ = std::vector<double>(n, 1.0);
+  }
+  else
+  {
+    // Check the size of the input weights
+    if (tmp_weights.size() != n)
+    {
+      ROS_WARN_STREAM_NAMED(LOG_NAMESPACE,
+                            "Incorrect number of weights ("
+                              << tmp_weights.size() << " provided, " << n
+                              << " expected). Using default weight vector (" << n << " x 1.0)");
+      cost_weights_ = std::vector<double>(n, 1.0);
+    }
```

Comment:
Suggestion: make this a fatal error.

If this is set incorrectly, it's a clear sign the user either didn't pay attention (in which case there could be more wrong with the configuration), the plugin is loading the wrong values, or the plugin expects a configuration which is different from what the user perhaps assumed she should provide.

Failing early would be good, as it provides a clear sign that something is wrong.